h2/src/main/org/h2/mvstore/RandomAccessStore.java

```diff
@@ -719,22 +718,31 @@ protected void doHousekeeping(MVStore mvStore) throws InterruptedException {
             });
         }
 
-        int chunksFillRate = getRewritableChunksFillRate();
-        int adjustedChunksFillRate = 100 - (100 - chunksFillRate) / 2;
-        int fillRateToCompare = isIdle() ? chunksFillRate : adjustedChunksFillRate;
+        int chunksFillRate = getChunksFillRate();
+        if (lastHousekeepingHasRewrittenChunks && chunksFillRate == previousChunksFillRate) {
```

Comment:
It looks like this condition may yield true only by a coincident, because if last housekeeping has rewritten chunks, chunks fill rate is likely to change, even without any user activity

Reply:
To be honest, It is based on the observations I have made on my database where practically the chunkFillRate was not moving during the housekeeping (quickly stayed in my case at 78%): Only the rewriteable chunk fill rate was incrementally increasing (up to 100%) until the retention time of the first rewritten chunks was hit, driving back the rewritable chunk fill rate to the starting  point and restarting the useless housekeeping cycle.
The condition has also an equality ( and not a <=) because I needed a way to detect when to "resume" the housekeeping if some user operations were made (at least enough operations to have an impact on the chunkFillRate) and without using isIdle.
_So the logic is really:_
1. The housekeeping has just rewritten some chunk with no impact on the chunkFillRate  ? -> Yes, then stop performing the  housekeeping
2. When to resume the housekeeping? -> Whenever there was enough user operations to change the chunkFillRate (by at least a point)

But you are right, I guess the fact that the chunkFillRate is not "moved" by the housekeeping in my tests might very well be specific to my database(s). And depending on the database, we could probably have a situation where chunkFillRate is also somehow fluctuating (78, 79, 77, 78, 79, ...) after each housekeeping, blowing up the stopping condition. 

I have seen you have made another proposal (in https://github.com/h2database/h2database/pull/4006). Thanks a lot for that. I will thoroughly test it and will come back to you as soon as I can.